website/content/en/status/report-2023-07-2023-09/squashfs.adoc

```diff
@@ -0,0 +1,29 @@
+=== SquashFS driver for FreeBSD kernel
+
+Links: +
+link:https://wiki.freebsd.org/SummerOfCode2023Projects/PortSquashFuseToTheFreeBSDKernel[Wiki page] URL: 
+link:https://wiki.freebsd.org/SummerOfCode2023Projects/PortSquashFuseToTheFreeBSDKernel[] +
+link:https://github.com/Mashijams/freebsd-src/tree/gsoc/testing[Source code] URL: https://github.com/Mashijams/freebsd-src/tree/gsoc/testing[]
+
+Contact: Raghav Sharma <raghav@FreeBSD.org>
+
+This quarter we finished SquashFS driver work for kernel. We now can mount SquashFS archives on the 13.2 or greater FreeBSD release and perform all basic 
+read-only filesystem operations.
+
+Code work includes:
+
+* Implementing vop_lookup() and vop_readdir() hooks for directory read support.
+* Implementing vop_read() and vop_strategy() hooks for files read support.
+* Implementing vop_readlink() hook for symlinks read support.
+
+We also implemented extended attributes interface functions for SquashFS and all the work remaining for them is to implement their kernel interface hooks.
+
+There were a lot of bug fixes as well, one major one was to find out why we can't list the first entry of the root directory, turns out SquashFS could have inode_number as zero 
+which kernel for some reason skips while listing dirents.
+For now, we fixed it by passing dummy inode_number to dirent instead of zero.
+
+The code review is currently ongoing with my mentor mailto:chuck@FreeBSD.org[Chuck Tuffli].
+
+I'm happy to say that SquashFS will find its place in upcoming FreeBSD releases.
```

Comment:
```suggestion
I am happy to say that SquashFS will find its place in upcoming FreeBSD releases.
```

Reply:
This formality is according to [a guideline for the FreeBSD Documentation Project](https://docs.freebsd.org/en/books/fdp-primer/book/#writing-style-guidelines).